### ISSUE TITLE
Is light-psi-all.jar still distributed with releases?

### ISSUE THREAD

**Issue by snadi** [JetBrains/Grammar-Kit#238]
Hi,
I'm trying to use the PSI classes in standalone applications and all the forms and suggestions point here towards using the light-psi-all.jar mentioned in this repo. However, I could not find that jar file with any of the releases. 

I tried to create it myself but for some reason, trying to build the project through gradle from within IntelliJ results in the following error:

```
Grammar-Kit/src/org/intellij/grammar/livePreview/LivePreviewHelper.java:105: error: no suitable constructor found for SingleAlarm(()->repars[...]ject),int,ThreadToUse,? super Project)
                          project -> new SingleAlarm(() -> reparseAllLivePreviews(project), 300, Alarm.ThreadToUse.SWING_THREAD, project));
                                     ^
    constructor SingleAlarm.SingleAlarm(Runnable,int,Disposable,ThreadToUse) is not applicable
      (argument mismatch; ThreadToUse cannot be converted to Disposable)
    constructor SingleAlarm.SingleAlarm(Runnable,int,ModalityState,Disposable) is not applicable
      (argument mismatch; ThreadToUse cannot be converted to ModalityState)
    constructor SingleAlarm.SingleAlarm(Runnable,int,ThreadToUse,Disposable) is not applicable
      (argument mismatch; ? super Project cannot be converted to Disposable)
```

even though building it from the command line using gradle works. Thus, I couldn't use the `lightpsi-all` run configurations from within IntelliJ.

Instead, after externally building the project using gradle, I then tried directly running the equivalent of that configuration from the command line (After creating a build-lightpsi directory):

```
java -ea -verbose:class -cp build/libs/grammar-kit-2020.1-SNAPSHOT.jar org.intellij.grammar.Main build-lightpsi grammars/* testData/generator/* build-lightpsi/grammars/* /Projects/all-grammars-test/grammars/* > build-lightpsi/classes.log.txt
```

but it resulted in the following error:

```
Error: Unable to initialize main class org.intellij.grammar.Main
Caused by: java.lang.NoClassDefFoundError: com/intellij/openapi/util/KeyedExtensionCollector
```

I'm assuming that's because I didn't provide the required jar files from the MANIFEST.MF mentioned in the ReadMe. However, I'm not sure where to find these jar files. I could find the annotations-19.0.0.jar already downloaded in my local maven repository for example, but I couldn't find the others and can't see any explicit dependencies declared in the build.gradle file so I can figure out which artifacts these jars belong to. I did try to search for them on the mvnrepository.com but for some of them, it's not clear which artifact is that.

Any help into getting the light-psi-all.jar would be appreciated!

Thanks.

**Comment by sischnei** [comment]
I'm running into the same issues you have described. I ended up getting a light-psi-all.jar in the binaries/ folder, but it seems it's incomplete or for some other reason not working. There is also this [one binary version online](http://www.java2s.com/Code/Jar/l/Downloadlightpsialljar.htm), but it seems out-of-date and differs quite a lot from the one I was able to generate.

It's a shame that grammar-kit doesn't seem to see a use-case for standalone usage of the generated parser and I haven't seen anybody really succeeding to do so (looking through the IntelliJ forums, there are some people who had mixed success). 

I think I will try to re-implement my grammar using ANTLR4 and try to get it integrated as an IntelliJ plugin. Hopefully, I'll have more success that way.

**Comment by gregsh** [comment]
`light-psi-all.jar` has always been a hack and just a proof-of-concept. 

Currently one can use [gradle-grammar-kit-plugin](https://github.com/JetBrains/gradle-grammar-kit-plugin) if one does not care for [that problem](https://github.com/JetBrains/gradle-grammar-kit-plugin/issues/23) which is way too much for me currently.

> It's a shame that grammar-kit doesn't seem to see a use-case

Apparently, I don't. Please share.

**Comment by jochenwierum** [comment]
For a small and personal project in my free time I use grammar-kit (you have been warned!). The project runs as a standalone application and consumes a custom domain specific language. I wrote an IntelliJ-Plugin to edit these DSL files. My goal was to use the same grammar and the same parser in the IntelliJ-Plugin and in the standalone application. This is why I found `light-psi-all.jar` useful.

I patched the Grammar-Kit build to produce a `light-psi-all.jar` that works for my purpose. Perhaps someone can use it as a starting point: https://github.com/jochenwierum/Grammar-Kit

However, I agree that the solution is a hack and my patch does not make it any better! ;)
